src/test/scala/org/broadinstitute/dsde/workbench/sam/openam/MockAccessPolicyDAO.scala

```diff
@@ -13,10 +13,37 @@ import scala.collection.mutable
 /**
   * Created by dvoet on 7/17/17.
   */
-class MockAccessPolicyDAO(private val policies: mutable.Map[WorkbenchGroupIdentity, WorkbenchGroup] = new TrieMap()) extends AccessPolicyDAO {
+class MockAccessPolicyDAO(private val resourceTypes: mutable.Map[ResourceTypeName, ResourceType], private val policies: mutable.Map[WorkbenchGroupIdentity, WorkbenchGroup] = new TrieMap()) extends AccessPolicyDAO {
```

Comment:
`MockAccessPolicyDAO` needs resourceTypes to have access to actions in roles. All these constructors made the refactoring easier